### ISSUE TITLE
Email working but email verification not sending

### ISSUE THREAD

**Issue by papa-gene** [getsentry/sentry#3766]
Hi all!

Not sure if this is an issue with Sentry, or some sort of configuration mistake on my part (was unable to find anything similar by googling).

I manage a Sentry 8.6.0 instance, installed by following the instructions on the "Installation Instructions/Python" page. I have configured outgoing email settings in /etc/sentry/config.yml. Upon entering the "Mail" section of the settings (on the web interface), I can successfully send a test email to myself. Email notifications are also being sent to users.

However, when I (or any user) go to the "Account" page, I am told my email hasn't been verified yet, and am given an option to resend a verification email. Clicking that pops a notification saying "a verification email has been sent to (my email address)". However, nothing is received. Checking the syslog log, the only message generated is: 

`Jul 21 16:06:18 HOSTNAME supervisord: sentry-web [DEBUG] root: Did not build any messages, no users to send to.`

The only non-standard feature I'm using is the [Sentry-LDAP](https://github.com/Banno/getsentry-ldap-auth) plugin, for SSO in my organization. It does however populate all fields of the "Account" page properly (including email).

Any pointers of what I could look for?

**Comment by mattrobenolt** [comment]
I'm not sure what sentry-ldap is doing (I've never looked at it myself), but that log message tells the story. This literally means that there were no email addresses for the email to get sent to, so it bailed out of attempting to send an email with no recipients. I'm not entirely sure why this would be the case for you though, but my guess is this is a bug in sentry-ldap not populating all of the right information.

**Comment by papa-gene** [comment]
Ok, I tried creating a local user, and sure enough, the "Resend verification email" worked.
So I went ahead and looked at the sentry table in postgresql, figured out that locally created users have an "email" entry in the "auth_user" table, and another "email" entry in "sentry_useremail". LDAP users only have the "auth_user" email, but not the "sentry_useremail". 

I'll contact the plugin creator to see if he can modify his plugin to populate the second field too.
Thanks for the help!
